### PR TITLE
feat: adding DD4hep digitization, fix measurement writer, add variance in o…

### PR DESCRIPTION
**Pull request by asalzburger**
This PR:
 * adds a DD4hep digitization example for debugging digitization issue
 * fixes one problem of `clusters` not being present when running smearing only
 * adds the variance to the `RootMeasurementWriter` in order to be able to do pulls/variance checks

odd-residual

odd-pull

### CODE REVIEW

**Review by noemina — APPROVED** [review]
Changes are fine. Approving.

File: Examples/Io/Root/include/ActsExamples/Io/Root/RootMeasurementWriter.hpp [hunk 1]
```diff
@@ -209,6 +211,19 @@ class RootMeasurementWriter final : public WriterT<MeasurementContainer> {
       recBound[Acts::eBoundPhi] = fullVect[Acts::eBoundPhi];
       recBound[Acts::eBoundTheta] = fullVect[Acts::eBoundTheta];
       recBound[Acts::eBoundTime] = fullVect[Acts::eBoundTime];
+
+      Acts::BoundSymMatrix fullVar =
+          m.expander() * m.covariance() * m.expander().transpose();
```
Comment (niermann999) [inline Examples/Io/Root/include/ActsExamples/Io/Root/RootMeasurementWriter.hpp:216]: For my understanding: Should the second one not just be the projection?

Comment (asalzburger) [reply Examples/Io/Root/include/ActsExamples/Io/Root/RootMeasurementWriter.hpp:216]: Ja, but it operates on a matrix.

Comment (niermann999) [reply Examples/Io/Root/include/ActsExamples/Io/Root/RootMeasurementWriter.hpp:216]: Ok, I just thought that .expander.transpose was be the same as .projection, with the latter being preferred

**Review by niermann999 — COMMENTED** [review]
approved

**Pull request merged**